### PR TITLE
Use cached thread pool for most AWS client operations

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
This updates nearly all of our AWS SDK clients to use a _cached_ rather than _fixed_ thread pool. Why is that desirable? We spotted that the application was falling over after approximately 24 hours. We eventually tracked this down to an OutOfMemoryException triggered by a lack of available threads. 

After much examination of the threading behaviour, we believe this happens because we are using the default `FixedThreadPool` that AWS provides in each of our clients, which has a maximum size of 10, and does not terminate idle threads whatsoever. Due to the behaviour of our application (large batches of network requests on a predictable schedule) we were observing that the thread pools slowly filled up over time and were never emptied of idle threads. Some of these threads were re-used, but it was not enough to stay under the limit we currently have on `MaxTasks` imposed by systemd (about 4900). This sounds like an awful lot, but we are creating thousands of SDK clients due to all the regions, accounts and services we work with.

`CachedThreadPool` on the other hand has no maximum size, but importantly it terminates idle threads after 60 seconds. This means that between scheduled tasks, we are able to remove idle threads, and stay at a much lower range of threads:

image

## What is the value of this?
It will keep the app running more consistently

## Will this require CloudFormation and/or updates to the AWS StackSet?
No

## Will this require changes to config?
No

## Any additional notes?
Big thanks for all the help from @jorgeazevedo on this one.

### CODE REVIEW

File: hq/app/aws/AWS.scala
Comment (jfsoul) [inline hq/app/aws/AWS.scala:55]: I just moved a few methods around so the logic in this file was easier to follow (`clients` next to where it was called from, for example)

**Comment by jfsoul** [comment]
This change was running in production over the weekend and the app was stable 🎉

**Review by jorgeazevedo — APPROVED** [review]
Great work on this, I think it's a neat and tidy solution!

**Comment by jorgeazevedo** [comment]
@jfsoul before you merge this, I was thinking of deploy the version current in main to gather some data and produce an identical chart, for a sort of before/after. I figure it might come in handy! I'd leave it collecting data from say 2pm today, and then collect it tomorrow morning (hopefully before the instance gets cycled).

**Comment by jfsoul** [comment]
Ok sure, I will hold off merging @jorgeazevedo

**Comment by jorgeazevedo** [comment]
@jfsoul I've disabled CD and will deploy an older main build in a couple of hours. Feel free to merge!

**Pull request merged**
